src/rest_vol.h

```diff
@@ -192,9 +192,79 @@
+        if (CURLE_OK != result)                                                                              \
+            FUNC_GOTO_ERROR(ERR_MAJOR, ERR_MINOR, ret_value, "%s", curl_easy_strerror(result));              \
+                                                                                                             \
+        if (handle_HTTP_response) {                                                                          \
+            long response_code;                                                                              \
+                                                                                                             \
+            if (CURLE_OK != curl_easy_getinfo(curl_ptr, CURLINFO_RESPONSE_CODE, &response_code))             \
+                FUNC_GOTO_ERROR(ERR_MAJOR, ERR_MINOR, ret_value, "can't get HTTP response code");            \
+                                                                                                             \
+            HANDLE_RESPONSE(response_code, ERR_MAJOR, ERR_MINOR, ret_value);                                 \
+        } /* end if */                                                                                       \
+    } while (0)
+
+/* Counterpart of CURL_PERFORM that takes a curl multi handle,
+ * and waits until all requests on it have finished before returning. */
+
+#define CURL_MULTI_PERFORM(curl_multi_ptr, ERROR_MAJOR, ERR_MINOR, ret_value)                                \
+    int      still_running = 0;                                                                              \
+    CURLMsg *curl_mult_msg = NULL;                                                                           \
+                                                                                                             \
+    do {                                                                                                     \
+        fprintf(stderr, "%zu transfers still running\n", still_running);                                     \
+        fprintf(stderr, "Performing...");                                                                    \
+        if (CURLM_OK != curl_multi_perform(curl_multi_ptr, &still_running))                                  \
+            FUNC_GOTO_ERROR(H5E_DATASET, H5E_WRITEERROR, ret_value, "cURL multi perform error");             \
+                                                                                                             \
+        if (still_running)                                                                                   \
+            /* wait for activity, timeout or "nothing"  */                                                   \
+            fprintf(stderr, "Polling...");                                                                   \
+        if (CURLM_OK != curl_multi_poll(curl_multi_ptr, NULL, 0, 1000, NULL))                                \
```

Comment:
I'm guessing you want to skip the entirety of this and the code below if `!still_running`? Otherwise it looks like a `curl_multi_poll` call is still made even if the multi transfer is finished.